geohisto/models.py

```diff
@@ -1,73 +1,70 @@
-from collections import OrderedDict, namedtuple
+import logging
 
-from .constants import DELTA, START_DATETIME
+from datetime import date
+from collections import OrderedDict, namedtuple, defaultdict
 
+from .constants import DELTA, START_DATETIME, END_DATE
+from .constants import END_KIND, END_NAME, END_INNER, END_TAXMODEL
+from .constants import INTERCOMMUNALITY_START_DATE
 
-class Towns(OrderedDict):
+log = logging.getLogger(__name__)
 
-    def upsert(self, town):
-        """Update or insert a Town, return True in case of insertion."""
-        is_created = town.id not in self
-        self[town.id] = town
+
+class Collection(object):
```

Comment:
I think it should inherit from `dict` or be named `CollectionMixin` and documented to be part of a class implementing the `dict` interface somehow.